src/content/tools/analyzer-plugins.md

```diff
@@ -0,0 +1,199 @@
+by enabling it in the `analysis_options.yaml` 
+file at the root of your package or workspace.
+
+### Enable a plugin
+
+To enable an analyzer plugin, add it to the top-level
+plugins section of your `analysis_options.yaml` file. You can
+add a plugin from [pub.dev][] using a version constraint:
+
+```yaml title="analysis_options.yaml"
+plugins:
+  my_plugin: ^1.0.0
+```
+
+Or, if you are developing a plugin locally, 
+you can use a local path:
+
+```yaml title="analysis_options.yaml"
+plugins:
+  my_plugin:
+    path: /path/to/my_plugin
+```
+
+After you have made changes to the plugins section, 
+restart the Dart Analysis Server to see the effects.
+
+[pub.dev]: {{site.pub}}
+
+### Configure diagnostics
+
+A plugin can report two kinds of diagnostics: warnings and lints. 
+The analyzer handles them differently by default:
+
+* Warnings that a plugin defines are enabled by 
+default (just like standard analyzer warnings).
+
+* Lint rules that a plugin defines are disabled by default 
+and must be explicitly enabled in your analysis options file.
+
+You can enable or disable specific lint rules under 
+the diagnostics section for a plugin:
+
+```yaml title="analysis_options.yaml"
+plugins:
+  my_plugin:
+    path: /path/to/my_plugin
+    diagnostics:
+      rule_1: true
+      rule_2: true
+      rule_3: false
+```
+
+In this example, `rule_1` and `rule_2` are enabled. 
+`rule_3` is explicitly disabled, which is useful for 
+overriding a rule that might be enabled from an `include:` file.
+
+### Suppress diagnostics
+
+You can suppress diagnostics from a plugin using the same 
+`ignore` comments as you would for [built-in Dart diagnostics][]. 
+To target a diagnostic from a specific plugin, 
+prefix the diagnostic code with the plugin name followed 
+by a slash (`/`).
+
+For example, to suppress a diagnostic with the 
+code `some_code` from a plugin named `some_plugin`, 
+use one of the following formats:
+
+To suppress for a single line:
+
+```dart
+// ignore: some_plugin/some_code
+```
+
+To suppress for an entire file:
+
+```dart
+// ignore_for_file: some_plugin/some_code
+```
+
+[built-in Dart diagnostics]: /tools/analysis#suppressing-diagnostics-for-a-file
+
+## Write an analyzer plugin
+
+Create your own analyzer plugin to add custom static analysis, 
+quick fixes, or assists to the Dart development experience.
+
+### Set up the plugin package
+
+An analyzer plugin is a standard Dart package with 
+specific dependencies and structure. The primary dependency 
+you will use is the [`analysis_server_plugin` package][], 
+which provides the core `Plugin` class.
+
+[`analysis_server_plugin` package]: {{site.pub-pkg}}/analysis_server_plugin
+
+1. Create the `pubspec.yaml` file:
+
+```yaml title"pubspec.yaml"
+name: test_analyzer_plugin
+version: 0.0.1
+
+environment:
+  sdk: ^3.10.0
+
+dependencies:
+  analysis_server_plugin: ^^0.3.0
+  analyzer: ^8.0.0
+```
+
+2. Create the plugin entry point
```

Comment:
I would restructure this so that 1. and 2. match in style. For example:

-------------

2. Create the plugin entry point. For example:

```dart title="lib/main.dart"
import 'package:analysis_server_plugin/plugin.dart';
import 'package:analysis_server_plugin/registry.dart';
final plugin = SimplePlugin();
class SimplePlugin extends Plugin {
  @override
  void register(PluginRegistry registry) {
    // Register diagnostics, quick fixes, and assists.
  }
}
```

The preceding code represents the basic structure for plugin entry point.
Make sure that you have a `lib/main.dart` file in your package because
this is the entry point for the analysis server to load your plugin.
```